azure/functions/decorators/function_app.py

```diff
@@ -176,8 +179,10 @@ def _validate_function(self) -> None:
                 f"Function {function_name} trigger {trigger} not present"
                 f" in bindings {bindings}")
 
-        if isinstance(trigger, HttpTrigger) and trigger.route is None:
-            trigger.route = self._function.get_function_name()
+        # Set route to function name if unspecified in the http trigger
+        if is_supported_trigger_type(trigger, HttpTrigger) \
+                and getattr(trigger, 'route', None) is None:
+            setattr(trigger, 'route', self._function.get_function_name())
```

Comment:
Why is setting the name in this code? This is one trigger-specific thing and I feel this could be route-specific. but let me know,